Description: Disable the JIT compiler on non-i386 architectures
 Upstream disables the JIT compiler on x86_64, PPC and ARMEL
 by default only which breaks any other architecture. This
 patch negates the #ifdef logic such that JIT is enabled
 on i386 only.

--- fs-uae-2.2.3+dfsg.orig/src/od-fs/defines.h
+++ fs-uae-2.2.3+dfsg/src/od-fs/defines.h
@@ -98,11 +98,8 @@ extern FILE *g_fs_uae_sync_debug_file;
 #define FPU_UAE
 #define GFXFILTER
 
-#if defined(__x86_64__) || defined(__ppc__) || defined(__ARMEL__)
-// no JIT for these architectures
-#elif defined(OPENBSD)
-// no sys/ucontext.h header
-#else
+//enable JIT on i386 only except OpenBSD
+#if defined(__i386__) && !defined(OPENBSD)
 #define JIT
 #define JIT_DEBUG
 #endif
